src/lasdi/latent_space.py

```diff
@@ -1,175 +1,482 @@
+        self.layers = torch.nn.ModuleList(self.layers)
+
+        # Now, initialize the weight matrices and bias vectors in the affine portion of each layer.
+        self.init_weight()
+
+        # Reshape input to the 1st layer or output of the last layer.
+        self.reshape_index : int        = reshape_index
+        self.reshape_shape : list[int]  = reshape_shape
+
+        # Set up the activation function. Note that we need to handle the threshold activation type
+        # separately because it requires an extra set of parameters to initialize.
+        self.act_type   : str               = act_type
         if act_type == "threshold":
-            self.act = act_dict[act_type](threshold, value)
-
-        elif act_type == "multihead":
-            self.use_multihead = True
-            if (self.n_layers > 3): # if you have more than one hidden layer
-                self.act = []
-                for i in range(self.n_layers-2):
-                    self.act += [act_dict[act_type](layer_sizes[i+1], num_heads)]
-            else:
-                self.act = [torch.nn.Identity()]  # No additional activation
-            self.act = torch.nn.ModuleList(self.fcs)
-
-        #all other activation functions initialized here
+            self.act    : torch.nn.Module   = act_dict[act_type](threshold, value)
         else:
-            self.act = act_dict[act_type]()
+            self.act    : torch.nn.Module   = act_dict[act_type]()
+
+        # All done!
         return
     
-    def forward(self, x):
+
+
+    def forward(self, x : torch.Tensor) -> torch.Tensor:
+        """
+        This function defines the forward pass through self.
+
+
+        -------------------------------------------------------------------------------------------
+        Arguments
+        -------------------------------------------------------------------------------------------
+
+        x: A tensor holding a batch of inputs. We pass this tensor through the network's layers 
+        and then return the result. If self.reshape_index == 0 and self.reshape_shape has k
+        elements, then the final k elements of x's shape must match self.reshape_shape. 
+
+
+        -------------------------------------------------------------------------------------------
+        Returns
+        -------------------------------------------------------------------------------------------
+
+        The image of x under the network's layers. If self.reshape_index == -1 and 
+        self.reshape_shape has k elements, then we reshape the output so that the final k elements
+        of its shape match those of self.reshape_shape.
+        """
+
+        # If the reshape_index is 0, we need to reshape x before passing it through the first 
+        # layer.
         if (self.reshape_index == 0):
-            # make sure the input has a proper shape
+            # Make sure the input has a proper shape. There is a lot going on in this line; let's
```

Comment:
I think most of these comments are somewhat repetitive of what is written earlier in arguments/returns section. I think we can reduced it into one sentence as below:
```
For k-dimension self.reshape_shape, make sure the last k dimension of x has the same shape.
```